File: problem_set1/me/mario.c
#include <cs50.h>
#include <stdio.h>

// Function prototypes
int get_height(void);
void print_pyramid(int height);

int main(void)
{
    // Get a valid height from the user
    int height = get_height();

    // Print the pyramid of the specified height
    print_pyramid(height);

    return 0;
}

// Function to prompt the user for a valid height
int get_height(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    } while (height < 1);
    return height;
}

// Function to print the pyramid
void print_pyramid(int height)
{
    for (int row = 1; row <= height; row++)
    {
        // Print the spaces
        for (int space = 0; space < height - row; space++)
        {
            printf(" ");
        }

        // Print the hashes
        for (int hash = 0; hash < row; hash++)
        {
            printf("#");
        }

        // Move to the next line
        printf("\n");
    }
}

File: problem_set6/sentimental-mario-less/mario.py
def main():
    height = get_height()
    pyramid(height)

def get_height():
    while True:
        try:
            height = int(input("Height: "))
            if 1 <= height <= 8:
                return height
        except ValueError:
            pass

def pyramid(height):
    for i in range(1, height + 1):
        spaces = " " * (height - i)
        hashes = "#" * i
        print(spaces + hashes)

if __name__ == "__main__":
    main()
